File: crates/fixtures/sample.c
/* Sample C module for testing */
#include <stdio.h>
#include <stdlib.h>

int helper(int x) {
    return x * 2;
}

int process(int *items, int len) {
    /* FIXME: check for null pointer */
    int total = 0;
    for (int i = 0; i < len; i++) {
        total += helper(items[i]);
    }
    return total;
}

File: crates/fixtures/sample.rs
//! Sample Rust module for testing.

use std::fs;
use std::path::Path;

/// Helper doubles a value.
pub fn helper(x: i32) -> i32 {
    x * 2
}

/// Process a list of items. Returns 0 for empty slices.
pub fn process(items: &[i32]) -> i32 {
    if items.is_empty() {
        return 0;
    }
    let mut total = 0;
    for item in items {
        total += helper(*item);
    }
    total
}
